cdap-client/src/main/java/co/cask/cdap/client/PreferencesClient.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.client;
+
+import co.cask.cdap.client.config.ClientConfig;
+import co.cask.cdap.client.exception.UnAuthorizedAccessTokenException;
+import co.cask.cdap.client.util.RESTClient;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.common.http.HttpMethod;
+import co.cask.common.http.HttpResponse;
+import co.cask.common.http.ObjectResponse;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Map;
+
+/**
+ * Provides ways to get/set Preferences.
+ */
+public class PreferencesClient {
+
+  private static final Gson GSON = new Gson();
+
+  private final RESTClient restClient;
+  private final ClientConfig config;
+
+  @Inject
+  public PreferencesClient(ClientConfig config) {
+    this.config = new ClientConfig.Builder(config).setApiVersion(Constants.Gateway.API_VERSION_3_TOKEN).build();
```

Comment:
maybe since you are creating a new ClientConfig here, it would be ok to set the apiVersion to v3 like you are doing, without resetting it. But please verify it when you write the CLI.